raster/r.what/main.c

```diff
@@ -472,50 +509,106 @@ int main(int argc, char *argv[])
+                        cache[point].north_buf, fs, cache[point].lab_buf);
+
+                for (i = 0; i < nfiles; i++) {
+                    if (out_type[i] == CELL_TYPE) {
+                        if (Rast_is_c_null_value(&cache[point].value[i])) {
+                            fprintf(stdout, "%s%s", fs, null_str);
+                            if (flg.label->answer)
+                                fprintf(stdout, "%s", fs);
+                            if (flg.color->answer)
+                                fprintf(stdout, "%s", fs);
+                            continue;
+                        }
+                        fprintf(stdout, "%s%ld", fs,
+                                (long)cache[point].value[i]);
+                        cache[point].dvalue[i] = cache[point].value[i];
+                    }
+                    else { /* FCELL or DCELL */
+
+                        if (Rast_is_d_null_value(&cache[point].dvalue[i])) {
+                            fprintf(stdout, "%s%s", fs, null_str);
+                            if (flg.label->answer)
+                                fprintf(stdout, "%s", fs);
+                            if (flg.color->answer)
+                                fprintf(stdout, "%s", fs);
+                            continue;
+                        }
+                        if (out_type[i] == FCELL_TYPE)
+                            sprintf(tmp_buf, "%.7g", cache[point].dvalue[i]);
+                        else /* DCELL */
+                            sprintf(tmp_buf, "%.15g", cache[point].dvalue[i]);
+                        G_trim_decimal(tmp_buf); /* not needed with %g? */
+                        fprintf(stdout, "%s%s", fs, tmp_buf);
+                    }
+                    if (flg.label->answer)
+                        fprintf(stdout, "%s%s", fs,
+                                Rast_get_d_cat(&(cache[point].dvalue[i]),
+                                               &cats[i]));
+                    if (flg.color->answer)
+                        fprintf(stdout, "%s%s", fs, cache[point].clr_buf[i]);
+                }
+                fprintf(stdout, "\n");
             }
-            fprintf(stdout, "%s%s%s%s%s", cache[point].east_buf, fs,
-                    cache[point].north_buf, fs, cache[point].lab_buf);
+            else {
+                point_value = json_value_init_object();
+                point_object = json_object(point_value);
 
-            for (i = 0; i < nfiles; i++) {
-                if (out_type[i] == CELL_TYPE) {
-                    if (Rast_is_c_null_value(&cache[point].value[i])) {
-                        fprintf(stdout, "%s%s", fs, null_str);
+                if (flg.cat->answer) {
+                    json_object_set_number(point_object, "cat",
+                                           cache[point].cat);
+                }
+
+                json_object_set_string(point_object, "easting",
+                                       cache[point].east_buf);
```

Comment:
Shouldn't this be converted to a number?

Reply:
It is a character field so I set is a string, should I convert it into a number using say `atof` and then set it as a number?

Reply:
Yes, thank you!